GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -552,7 +552,15 @@ private void UICommands_PostRepositoryChanged(object sender, GitUIEventArgs e)
         {
             // Note that this called in most FormBrowse context to "be sure"
             // that the repo has not been updated externally.
-            RefreshRevisions(e.GetRefs);
+
+            // It can also be called from background tasks, e.g. from BackgroundFetchPlugin.
+            if (ThreadHelper.JoinableTaskContext.IsOnMainThread)
+            {
+                RefreshRevisions(e.GetRefs);
+                return;
+            }
+
+            Invoke(() => RefreshRevisions(e.GetRefs));
```

Comment:
What do you think about inverting this?
```suggestion
            // It can also be called from background tasks, e.g. from BackgroundFetchPlugin.
            if (!ThreadHelper.JoinableTaskContext.IsOnMainThread)
            {
                Invoke(() => RefreshRevisions(e.GetRefs));
                return;
            }

            RefreshRevisions(e.GetRefs);
```

Reply:
Both are OK with me; positive test versus error exit flow

Reply:
I thought about it and quickly decided to avoid the negation `!` as usual. But in this case, it reads better with the negation.
Done.